packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -289,7 +290,22 @@ export function RepositoryMixin<T extends MixinTarget<Application>>(
 
         if (operation in ds && typeof ds[operation] === 'function') {
           debug('Migrating dataSource %s', b.key);
-          await ds[operation](options.models);
+          if (!options?.models || options?.models.length === 0) {
+            // if options.models is an empty array, proceed normally
+            await ds[operation](options.models);
+          } else {
+            // if options.models is not empty, check if model is
+            // connected to the datasource
+            const models = [];
+            for (const model of options?.models) {
+              // todo : add "definitions" attribute in Datasource definition
+              // in "loopback-datasource-juggler"
+              if (Object.keys(ds['definitions']).includes(model)) {
```

Comment:
> There's an attribute that needs to be added in DataSource model in "loopback-datasource-juggler"

Here is the place where to add type declarations for `definitions` property:

https://github.com/strongloop/loopback-datasource-juggler/blob/8d1690e9b2c76e3d0a8336b47f133e0605cbcc9a/types/datasource.d.ts#L77-L85

Reply:
i have to create a PR to fix it?

Reply:
Yes please!

Reply:
I see you opened https://github.com/strongloop/loopback-datasource-juggler/pull/1844 👏🏻